helper/resource/testing_new_import_block_id_test.go

```diff
@@ -0,0 +1,459 @@
+			},
+		},
+	})
+}
+
+func TestTest_TestStep_ImportBlockId_SkipDataSourceState(t *testing.T) {
+	t.Parallel()
+
+	UnitTest(t, TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version1_5_0), // ProtoV6ProviderFactories
+		},
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"examplecloud": providerserver.NewProviderServer(testprovider.Provider{
+				DataSources: map[string]testprovider.DataSource{
+					"examplecloud_thing": {
+						ReadResponse: &datasource.ReadResponse{
+							State: tftypes.NewValue(
+								tftypes.Object{
+									AttributeTypes: map[string]tftypes.Type{
+										"id": tftypes.String,
+									},
+								},
+								map[string]tftypes.Value{
+									"id": tftypes.NewValue(tftypes.String, "datasource-test"),
+								},
+							),
+						},
+						SchemaResponse: &datasource.SchemaResponse{
+							Schema: &tfprotov6.Schema{
+								Block: &tfprotov6.SchemaBlock{
+									Attributes: []*tfprotov6.SchemaAttribute{
+										{
+											Name:     "id",
+											Type:     tftypes.String,
+											Computed: true,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Resources: map[string]testprovider.Resource{
+					"examplecloud_thing": {
+						CreateResponse: &resource.CreateResponse{
+							NewState: tftypes.NewValue(
+								tftypes.Object{
+									AttributeTypes: map[string]tftypes.Type{
+										"id": tftypes.String,
+									},
+								},
+								map[string]tftypes.Value{
+									"id": tftypes.NewValue(tftypes.String, "resource-test"),
+								},
+							),
+						},
+						ImportStateResponse: &resource.ImportStateResponse{
+							State: tftypes.NewValue(
+								tftypes.Object{
+									AttributeTypes: map[string]tftypes.Type{
+										"id": tftypes.String,
+									},
+								},
+								map[string]tftypes.Value{
+									"id": tftypes.NewValue(tftypes.String, "resource-test"),
+								},
+							),
+						},
+						SchemaResponse: &resource.SchemaResponse{
+							Schema: &tfprotov6.Schema{
+								Block: &tfprotov6.SchemaBlock{
+									Attributes: []*tfprotov6.SchemaAttribute{
+										{
+											Name:     "id",
+											Type:     tftypes.String,
+											Computed: true,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}),
+		},
+		Steps: []TestStep{
+			{
+				Config: `
+					data "examplecloud_thing" "test" {}
+					resource "examplecloud_thing" "test" {}
+				`,
+			},
+			{
+				ResourceName:    "examplecloud_thing.test",
+				ImportState:     true,
+				ImportStateKind: ImportBlockWithId,
+				ImportStateCheck: func(is []*terraform.InstanceState) error {
+					if len(is) > 1 {
+						return fmt.Errorf("expected 1 state, got: %d", len(is))
+					}
+
+					return nil
+				},
+			},
+		},
+	})
+}
+
+func TestTest_TestStep_ImportBlockId_ImportStateVerifyIgnore_Real_Example(t *testing.T) {
```

Comment:
On the fence about how useful this actually is for testing the behaviour of `ImportStateVerifyIgnore`. Considering removing it before this goes in.